Warnings in examples

### Describe the bug

Compile warnings in the github workflow.

### Steps to reproduce

Just compile example code mqtt_auth, mqtt_progmem, mqtt_publish_in_callback, mqtt_reconnect_nonblocking, mqtt_stream

### Expected behavior

No warning.

### Library version

v3.2.1

### Platform

Arduino AVR

### Environment

```markdown
- Board: Arduino Uno
- Framework Version: Arduino AVR 5.2.0
- Development Environment: Arduino-cli
```

### Minimal reproducible example (MVP)

```markdown

```

### Relevant log output

```shell
Compiling sketch: examples/mqtt_progmem
  /home/runner/work/pubsubclient3/pubsubclient3/examples/mqtt_progmem/src/mqtt_progmem.cpp: In function 'void callback(char*, uint8_t*, size_t)':
  /home/runner/work/pubsubclient3/pubsubclient3/examples/mqtt_progmem/src/mqtt_progmem.cpp:22:21: warning: unused parameter 'topic' [-Wunused-parameter]
   void callback(char* topic, uint8_t* payload, size_t plength) {
                       ^~~~~
  /home/runner/work/pubsubclient3/pubsubclient3/examples/mqtt_progmem/src/mqtt_progmem.cpp:22:37: warning: unused parameter 'payload' [-Wunused-parameter]
   void callback(char* topic, uint8_t* payload, size_t plength) {
                                       ^~~~~~~
  /home/runner/work/pubsubclient3/pubsubclient3/examples/mqtt_progmem/src/mqtt_progmem.cpp:22:53: warning: unused parameter 'plength' [-Wunused-parameter]
   void callback(char* topic, uint8_t* payload, size_t plength) {
                                                       ^~~~~~~
  Sketch uses 14900 bytes (46%) of program storage space. Maximum is 32256 bytes.
  Global variables use 372 bytes (18%) of dynamic memory, leaving 1676 bytes for local variables. Maximum is 2048 bytes.
```

--- a/examples/mqtt_auth/mqtt_auth.ino
+++ b/examples/mqtt_auth/mqtt_auth.ino
@@ -11,6 +11,8 @@
 #include <PubSubClient.h>
 #include <SPI.h>
 
+#define _UNUSED_ __attribute__((unused))
+
 // Update these with values suitable for your network.
 byte mac[] = {0xDE, 0xED, 0xBA, 0xFE, 0xFE, 0xED};
 IPAddress ip(172, 16, 0, 100);

--- a/examples/mqtt_publish_in_callback/mqtt_publish_in_callback.ino
+++ b/examples/mqtt_publish_in_callback/mqtt_publish_in_callback.ino
@@ -17,6 +17,8 @@
 #include <PubSubClient.h>
 #include <SPI.h>
 
+#define _UNUSED_ __attribute__((unused))
+
 // Update these with values suitable for your network.
 byte mac[] = {0xDE, 0xED, 0xBA, 0xFE, 0xFE, 0xED};
 IPAddress ip(172, 16, 0, 100);

--- a/examples/mqtt_reconnect_nonblocking/mqtt_reconnect_nonblocking.ino
+++ b/examples/mqtt_reconnect_nonblocking/mqtt_reconnect_nonblocking.ino
@@ -11,6 +11,8 @@
 #include <PubSubClient.h>
 #include <SPI.h>
 
+#define _UNUSED_ __attribute__((unused))
+
 // Update these with values suitable for your hardware/network.
 byte mac[] = {0xDE, 0xED, 0xBA, 0xFE, 0xFE, 0xED};
 IPAddress ip(172, 16, 0, 100);

--- a/examples/mqtt_stream/mqtt_stream.ino
+++ b/examples/mqtt_stream/mqtt_stream.ino
@@ -14,6 +14,8 @@
 #include <SPI.h>
 #include <SRAM.h>
 
+#define _UNUSED_ __attribute__((unused))
+
 // Update these with values suitable for your network.
 byte mac[] = {0xDE, 0xED, 0xBA, 0xFE, 0xFE, 0xED};
 IPAddress ip(172, 16, 0, 100);

